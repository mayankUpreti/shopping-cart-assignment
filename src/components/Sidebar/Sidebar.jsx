import React,{useState,useEffect} from "react";
import { NavLink,useHistory } from "react-router-dom";
import { fetchcategories } from "../../api/api";

const Sidebar=({category,setCategory})=>{
  const [categories,setCategories]=useState([]);
  useEffect(async()=>{
    const response=await fetchcategories();
    setCategories(response)
},[])

  return (
    <>
   
      <div className="nav flex-column nav-pills sidebar">
        {
          categories && categories.map(el=>
            <div
            key={el.id}
            onClick={()=>setCategory(el.id)}
            className={`nav-link p-3 ${category===el.id?'active':''}`}
            // activeClassName="active"
          >
         
            {el.name}
          </div>
            )
        }
    </div>
    </>
  );
}

export default Sidebar;
