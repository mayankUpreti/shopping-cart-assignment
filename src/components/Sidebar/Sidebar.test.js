import React  from 'react'
import Sidebar from './Sidebar'
import {shallow} from 'enzyme';

it('testing a custom Sidebar ',()=>{
    const category='Beverages';
    const setCategory=jest.fn()
    expect(shallow(<Sidebar category={category} setCategory={setCategory}/>)).toMatchSnapshot()
})