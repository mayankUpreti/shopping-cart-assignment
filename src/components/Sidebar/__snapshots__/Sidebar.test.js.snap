// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing a custom Sidebar  1`] = `
<Fragment>
  <div
    aria-label="categories"
    className="nav flex-column nav-pills sidebar h-100"
    tabIndex={0}
  />
</Fragment>
`;
